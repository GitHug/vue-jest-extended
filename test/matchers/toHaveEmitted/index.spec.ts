import FakeWrapper from '../../utils';
import matcher from '../../../src/matchers/toHaveEmitted';

expect.extend(matcher);

class Dummy {
  foo = 'bar';
}

describe('.toHaveEmitted', () => {
  const wrapper = new FakeWrapper();
  wrapper.emit('input', { data: 'test' });
  wrapper.emit('change', { test: { data: new Dummy() } });
  wrapper.emit('keyUp');

  test('that toHaveEmittedPayload is an alias for toHaveEmitted', () => {
    expect(matcher.toHaveEmittedPayload).toBe(matcher.toHaveEmitted);
  });

  describe('passes when', () => {
    test.each(['input', 'change', 'keyUp'])(
      'the expected event %s matches one emitted by the wrapper',
      (event: string) => {
        expect(wrapper).toHaveEmitted(event);
      },
    );

    test.each([
      ['input', { data: 'test' }],
      ['change', { test: { data: new Dummy() } }],
    ])(
      'the expected event %s and the expected payload %o matches event and payload emitted',
      (event: string, payload: any) => {
        expect(wrapper).toHaveEmitted(event, payload);
      },
    );
  });

  describe('fails when', () => {
    test.each([undefined, null, NaN, [], {}, 'keyDown'])(
      'the expected event %s has not been emitted by the wrapper',
      (event: any) => {
        expect(() => {
          expect(wrapper).toHaveEmitted(event);
        }).toThrowErrorMatchingSnapshot();
      },
    );

    test.each([
      ['input', { test: 'data' }],
      ['change', { test: { data: [{}] } }],
    ])('the expected event %s and payload %o combination has not been emitted', (event: string, payload: any) => {
      expect(() => {
        expect(wrapper).toHaveEmitted(event, payload);
      }).toThrowErrorMatchingSnapshot();
    });
  });
});
