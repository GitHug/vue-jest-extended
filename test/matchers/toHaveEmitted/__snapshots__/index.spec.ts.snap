// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.toHaveEmitted fails when the expected event [] has not been emitted by the wrapper 1`] = `
"
  [2mexpect([22m[31mreceived[39m[2m).toHaveEmitted([22m[32mexpected[39m[2m)[22m


  Expected wrapper to have emitted:

    [32m[][39m

  received:
    [31m\\"input, change, keyUp\\"[39m
  "
`;

exports[`.toHaveEmitted fails when the expected event {} has not been emitted by the wrapper 1`] = `
"
  [2mexpect([22m[31mreceived[39m[2m).toHaveEmitted([22m[32mexpected[39m[2m)[22m


  Expected wrapper to have emitted:

    [32m{}[39m

  received:
    [31m\\"input, change, keyUp\\"[39m
  "
`;

exports[`.toHaveEmitted fails when the expected event NaN has not been emitted by the wrapper 1`] = `
"
  [2mexpect([22m[31mreceived[39m[2m).toHaveEmitted([22m[32mexpected[39m[2m)[22m


  Expected wrapper to have emitted:

    [32mNaN[39m

  received:
    [31m\\"input, change, keyUp\\"[39m
  "
`;

exports[`.toHaveEmitted fails when the expected event change and payload { test: { data: [ {}, [length]: 1 ] } } combination has not been emitted 1`] = `
"
  [2mexpect([22m[31mreceived[39m[2m).toHaveEmitted([22m[32mexpected[39m[2m)[22m


  Expected wrapper to have emitted:

    [32m{\\"change\\": {\\"test\\": {\\"data\\": [{}]}}}[39m

  received:
    [31m{\\"test\\": {\\"data\\": {\\"foo\\": \\"bar\\"}}}[39m
  "
`;

exports[`.toHaveEmitted fails when the expected event input and payload { test: 'data' } combination has not been emitted 1`] = `
"
  [2mexpect([22m[31mreceived[39m[2m).toHaveEmitted([22m[32mexpected[39m[2m)[22m


  Expected wrapper to have emitted:

    [32m{\\"input\\": {\\"test\\": \\"data\\"}}[39m

  received:
    [31m{\\"data\\": \\"test\\"}[39m
  "
`;

exports[`.toHaveEmitted fails when the expected event keyDown has not been emitted by the wrapper 1`] = `
"
  [2mexpect([22m[31mreceived[39m[2m).toHaveEmitted([22m[32mexpected[39m[2m)[22m


  Expected wrapper to have emitted:

    [32m\\"keyDown\\"[39m

  received:
    [31m\\"input, change, keyUp\\"[39m
  "
`;

exports[`.toHaveEmitted fails when the expected event null has not been emitted by the wrapper 1`] = `
"
  [2mexpect([22m[31mreceived[39m[2m).toHaveEmitted([22m[32mexpected[39m[2m)[22m


  Expected wrapper to have emitted:

    [32mnull[39m

  received:
    [31m\\"input, change, keyUp\\"[39m
  "
`;

exports[`.toHaveEmitted fails when the expected event undefined has not been emitted by the wrapper 1`] = `
"
  [2mexpect([22m[31mreceived[39m[2m).toHaveEmitted([22m[32mexpected[39m[2m)[22m


  Expected wrapper to have emitted:

    [32mundefined[39m

  received:
    [31m\\"input, change, keyUp\\"[39m
  "
`;
